state/secrets.go

```diff
@@ -1406,33 +1405,46 @@ func (st *State) removeRemoteSecretConsumer(appName string) error {
 
 	match := fmt.Sprintf("(unit|application)-%s(\\/\\d)?", appName)
 	q := bson.D{{"consumer-tag", bson.D{{"$regex", match}}}}
-	iter := secretConsumersCollection.Find(q).Select(bson.D{{"consumer-tag", 1}}).Iter()
+	_, err := secretConsumersCollection.Writeable().RemoveAll(q)
+	return err
```

Comment:
So we don't need to remove refCount anymore here?

Reply:
Remote secret consumers exist in the offering model don't use labels. A follow up PR will introduce a separate collection to store these records.